programs/local/LocalServer.cpp

```diff
@@ -323,12 +323,30 @@ void LocalServer::setupUsers()
     auto & access_control = global_context->getAccessControl();
     access_control.setNoPasswordAllowed(config().getBool("allow_no_password", true));
     access_control.setPlaintextPasswordAllowed(config().getBool("allow_plaintext_password", true));
-    if (config().has("users_config") || config().has("config-file") || fs::exists("config.xml"))
+    if (config().has("config-file") || fs::exists("config.xml"))
     {
-        const auto users_config_path = config().getString("users_config", config().getString("config-file", "config.xml"));
-        ConfigProcessor config_processor(users_config_path);
-        const auto loaded_config = config_processor.loadConfig();
-        users_config = loaded_config.configuration;
+        String config_path = config().getString("config-file","");
+        bool has_user_directories = config().has("user_directories");
+        const auto config_dir = std::filesystem::path{config_path}.remove_filename().string();
+        String users_config_path = config().getString("users_config","");
+        if (users_config_path.empty() && !has_user_directories)
+            users_config = getConfigurationFromXMLString(minimal_default_user_xml);
```

Comment:
why not just like this?
```suggestion
        if (users_config_path.empty())
            users_config = getConfigurationFromXMLString(minimal_default_user_xml);
```

Reply:
hmm.. if user defined config-file option but config file doesn't have  <users_config> and <users_directories> Then we are  good to define minimal_default_user . 
OR 
--config-option is not defined i.e.. config().has("config-file", "") is empty , then also we should add minimal_default_user.
Please correct me if I am wrong .